### PR TITLE
Expose OIDC audience of KafkaChannel in its status

### LINKED ISSUE #3576: [control-plane]: Support exposing the Audience of a KafkaChannel
As the [Eventing OIDC feature track](https://docs.google.com/document/d/1e7UgNTkL0Br5Da09Rg2ieVmhKJo4VXuBj-mHT9NCujY/edit#heading=h.glanb1rq6iu5) describes, Addressables (and thus KafkaChannels) need to announce their audience in `.status.address.audience` in the format `GROUP/KIND/NAMESPACE/NAME` (e.g. `messaging.knative.dev/kafkachannel/myns/mykc`).

In particular the KafkaChannel controller should:

* when the feature flag (see https://github.com/knative/eventing/issues/7174) is disabled:
  * no change in behavior
* when the feature flag (see https://github.com/knative/eventing/issues/7174) is enabled:
  * Set the KafkaChannels `.status.address.audience` accordingly.

Additional Information:
* Similar to https://github.com/knative/eventing/issues/7187

### PR DESCRIPTION
**Pull request by creydr**
Fixes #3576

## Proposed Changes

- :gift: Provision a KafkaChannels `.status.address.audience` and `.status.addresses[*].audience` field
- :broom: Update KafkaChannel CRD to align with latest SubscribableSpec and SubscribableStatus (include `replyAudience`, `subscriberAudience`, `auth.serviceAccountName`)

```
$ kubectl get kafkachannel my-kafka-channel -o yaml  
apiVersion: messaging.knative.dev/v1beta1
kind: KafkaChannel
metadata:
  name: my-kafka-channel
  namespace: default
  ...
spec:
  ...
status:
  address:
    audience: messaging.knative.dev/kafkachannel/default/my-kafka-channel
    name: http
    url: http://my-kafka-channel-kn-channel.default.svc.cluster.local
  addresses:
  - audience: messaging.knative.dev/kafkachannel/default/my-kafka-channel
    name: http
    url: http://my-kafka-channel-kn-channel.default.svc.cluster.local
...
```

**Release Note**

```release-note
Expose the OIDC Audience of a KafkaChannel in its status
```

### CODE REVIEW

**Comment by creydr** [comment]
@pierDipi this PR is ready for review

**Comment by creydr** [comment]
Rebased after update-deps PR merged

File: control-plane/pkg/reconciler/channel/v2/channelv2.go [hunk 1]
```diff
@@ -307,11 +319,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, channel *messagingv1beta
 			return err
 		}
 
-		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, audience, channelService, caCerts)
 		addressableStatus.Addresses = []duckv1.Addressable{httpsAddress}
 		addressableStatus.Address = &httpsAddress
 	} else {
-		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost)
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, audience)
```
Comment (Leo6Leo) [inline control-plane/pkg/reconciler/channel/v2/channelv2.go:326]: If this is not in TLS mode, I think we may not need audience here?

File: control-plane/pkg/reconciler/channel/v2/channelv2.go [hunk 2]
```diff
@@ -278,26 +281,35 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, channel *messagingv1beta
...
-	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+	if featureFlags.IsPermissiveTransportEncryption() {
 		caCerts, err := r.getCaCerts()
 		if err != nil {
 			return err
 		}
 
-		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost)
-		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, audience)
```
Comment (Leo6Leo) [inline control-plane/pkg/reconciler/channel/v2/channelv2.go:303]: If this is not in TLS mode, I think we may not need audience here?

File: control-plane/pkg/reconciler/channel/v2/channelv2.go (hunk 1 above)
Comment (creydr) [reply control-plane/pkg/reconciler/channel/v2/channelv2.go:326]: We can still use it. OIDC auth is not secure without TLS but it's not a requirement

File: control-plane/pkg/reconciler/channel/v2/channelv2.go (hunk 2 above)
Comment (creydr) [reply control-plane/pkg/reconciler/channel/v2/channelv2.go:303]: Same as above

**Pull request merged**